01-basics/01-CounterButtonApp/script.js

```diff
@@ -1,3 +1,21 @@
-// import { createApp } from './vendor/vue.esm-browser.js';
+import { createApp, defineComponent } from './vendor/vue.esm-browser.js';
 
-// Создайте Vue приложение
+const app = createApp({});
+
+const Counter = defineComponent({
+  data() {
+    return {
+      counter: 0,
+    };
+  },
+  methods: {
+    increment() {
+      this.counter++;
+    },
+  },
+  template: '<button type="button" @click="increment">{{ counter }}</button>',
+});
+
+app.component('counter', Counter);
+
```

Comment:
I'd prefer local registration of component instead of global one. It's more explicit, you exactly see what child component actually is.